fwupd-fixed-more

etbe opened the pull request.

dsugar100 commented on policy/modules/system/fwupd.te:
```diff
@@ -0,0 +1,261 @@
+dbus_read_lib_files(fwupd_t)
+dbus_system_bus_client(fwupd_t)
+dbus_connect_system_bus(fwupd_t)
+dbus_use_system_bus_fds(fwupd_t)
```
Maybe you want to also put these dbus_* uses in an optional block

etbe replied:
It uses dbus to communicate between the daemon and the utility to send commands and according to the results of google searches isn't really usable without dbus. Would it make sense to build a policy with fwupd but without dbus?

dsugar100 replied:
I understand.   Maybe it isn't an issue here.  Though below you do wrap the dbus interface uses in optional (fwupdmgr_t).  And I understand some distributions don't have dbus, but maybe they also wouldn't be using fwudpd either.  I'm ok with it as is.

etbe commented:
I think the changes I pushed to that PR have addressed all the issues you 
raised.

On Tuesday, 17 June 2025 04:01:06 AEST Chris PeBenito wrote:
> @pebenito requested changes on this pull request.
> 
> > @@ -0,0 +1,5 @@
> 
> +/usr/bin/fwupdmgr	--	
gen_context(system_u:object_r:fwupdmgr_exec_t,s0)
> +/usr/libexec/fwupd/fwupd --	gen_context(system_u:object_r:fwupd_exec_t,s0)
> +/var/lib/fwupd(/.*)?		
gen_context(system_u:object_r:fwupd_var_lib_t,s0)
> +/var/cache/fwupd(/.*)?		
gen_context(system_u:object_r:fwupd_cache_t,s0)
> +/etc/fwupd(/.*)?		gen_context(system_u:object_r:fwupd_conf_t,s0)
> 
> Please move to top of file.
> 
> > +type fwupd_var_lib_t;
> 
> +files_type(fwupd_var_lib_t)
> +
> +type fwupd_cache_t;
> +files_type(fwupd_cache_t)
> +
> +type fwupd_conf_t;
> +files_type(fwupd_conf_t)
> +
> +type fwupdmgr_tmpfs_t;
> +files_tmpfs_file(fwupdmgr_tmpfs_t)
> +
> +type fwupd_runtime_t;
> +files_runtime_file(fwupd_runtime_t)
> 
> Please reorder in alphabetical order.
> 
> > @@ -2459,6 +2459,24 @@ interface(`dev_rw_framebuffer',`
> 
>  	rw_chr_files_pattern($1, device_t, framebuf_device_t)
>  ')
> 
> +########################################
> +## <summary>
> +##	Allow read the gpiochip device
> +## </summary>
> +## <param name="domain">
> +##	<summary>
> +##	Domain allowed access.
> +##	</summary>
> +## </param>
> +#
> +interface(`dev_read_gpiochip_device',`
> 
> Please remove `device` from the name here and also in the new `mei`
> interface.
> > +##	Map user runtime files.
> 
> +## </summary>
> +## <param name="domain">
> +##	<summary>
> +##	Domain allowed access.
> +##	</summary>
> +## </param>
> +#
> +interface(`userdom_map_user_runtime_files',`
> +	gen_require(`
> +		type user_runtime_t;
> +	')
> +
> +	allow $1 user_runtime_t:file map;
> +')
> +
> 
> Please merge these into a single `mmap_manage` interface.
> 
> > @@ -1003,6 +1003,24 @@ interface(`kernel_dontaudit_getattr_proc',`
> 
>  	dontaudit $1 proc_t:filesystem getattr;
>  ')
> 
> +########################################
> +## <summary>
> +##	Watch proc dirs.
> +## </summary>
> +## <param name="domain">
> +##	<summary>
> +##	Domain allowed access.
> +##	</summary>
> +## </param>
> +#
> +interface(`kernel_watch_proc',`
> 
> `kernel_watch_proc_dirs`
> 
> > @@ -2220,6 +2238,24 @@ interface(`kernel_dontaudit_search_kernel_sysctl',`
> 
>  	dontaudit $1 sysctl_kernel_t:dir search;
>  ')
> 
> +########################################
> +## <summary>
> +##	Allow watching sysctl kernel dirs
> +## </summary>
> +## <param name="domain">
> +##	<summary>
> +##	Domain to allow watching.
> +##	</summary>
> +## </param>
> +#
> +interface(`kernel_watch_kernel_sysctl',`
> 
> `kernel_watch_kernel_sysctl_dirs`


-- 
My Main Blog         http://etbe.coker.com.au/
My Documents Blog    http://doc.coker.com.au/

etbe commented:
On Tuesday, 17 June 2025 23:14:19 AEST Dave Sugar wrote:
> +dbus_read_lib_files(fwupd_t)
> +dbus_system_bus_client(fwupd_t)
> +dbus_connect_system_bus(fwupd_t)
> +dbus_use_system_bus_fds(fwupd_t)
> 
> I understand.   Maybe it isn't an issue here.  Though below you do wrap the
> dbus interface uses in optional (fwupdmgr_t).  And I understand some
> distributions don't have dbus, but maybe they also wouldn't be using fwudpd
> either.  I'm ok with it as is.

Good point consistency is important.  I've made them all optional.

-- 
My Main Blog         http://etbe.coker.com.au/
My Documents Blog    http://doc.coker.com.au/
